### ISSUE TITLE
Drush updb derails on Windows

### ISSUE THREAD

**Issue by Cswami1008** [drush-ops/drush#3763]
**Describe the bug**
I am running Windows-7 using the CMDer command line application.  Was trying to update a Drupal-8 Database on a development local site.  The "Drush updb" command ends with this error:

"[error]  The command could not be executed successfully (returned: 'C:/Users/Chandra' is not recognized as an internal or external command, operable program or batch file., code: 1) [13.17 sec, 38.98 MB]"

My guess is that since my User-Directory is "Chandra Swami" Drush is biting off the rest of the Path to the executable. The space between "Chandrda" and "Swami" has derailed the batch file.

**To Reproduce**
Nothing special.  I am on a Windows-7 machine and type: "Drush updb" into the command prompt and 13 seconds later I have this nice error.  I am using XAMPP.

**Expected behavior**
I would receive a Database update.

**Actual behavior**
The above mentioned error.

**Workaround**
If there is, I would like to know about it.  You see Composer is the NEW WAY in Drupal-8 and Composer cannot do Database Updates for itself.  If I do a Drupal-Core Update I am stuck because the Database is not Updated and this results in a Machine-Barf of unlimited SQL Errors.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 9.2.3
| Drupal version? |8.5.6
| PHP version     | 5.6.34
| OS?             |Windows

**Additional information**

**Comment by greg-1-anderson** [comment]
@Cswami1008 Could you please try https://github.com/drush-ops/drush/pull/3758 and report back as to whether a) it works better or worse for this particular issue, and b) whether it works better or worse for other commands.

**Comment by Cswami1008** [comment]
Still trying to learn how to use Git to apply a patch to Drupal-8.  My site is a local-development version of Drupal-8, Drush-9, Composer and PHP-7.2.  I am on a Windows-7-HP machine.

Sorry to be such a noob.

**Comment by Cswami1008** [comment]
[Issue in Drupal.org] (https://www.drupal.org/forum/support/post-installation/2018-10-29/catch-22-with-composer-and-drupal-8)

**Comment by Cswami1008** [comment]
I did this and received an error message.  InvalidArgumentException with options to load other packages with vaguely similar names.  However, the current version of Drush reads Drush-9.5.2.

Absolutely no change in the error when running the "drush updb" command.
ERROR-MESSAGE: (returned: Could not open input file: C:\ProgramData\drush\\drush.phar
, code: 1) [43.28 sec, 24.98 MB].

**Comment by greg-1-anderson** [comment]
Drush 9 is not packaged as a phar. Perhaps your error message is related to the Drush Launcher, which is a separate project with different maintainers.

What does Composer report when you run `composer info drush/drush`?

If you set your current working directory to the path where your composer.json is (the project root), you should be able to run `./vendor/bin/drush updatedb`. That should give you the exact behavior of Drush itself on Windows without any involvement from the launcher.

**Comment by Cswami1008** [comment]
I type:  php composer.phar info drush/drush

I get:  [InvalidArgumentException]
  Package drush/drush not found
-------------------------------------------------------------------------------
I type: composer info drush/drush

I get: 'composer' is not recognized as an internal or external command, operable program or batch file.
-------------------------------------------------------------------------------
I type: which drush

I get: /c/ProgramData/drush/drush
-------------------------------------------------------------------------------
When you say "./vendor/bin/drush updatedb", what is the entire command.  I tried many composer permutations and nothing worked.

**Comment by greg-1-anderson** [comment]
Since you said you were using Composer, I presumed that you had a site-local Drush. Did you build your site using [drupal-composer/drupal-project](https://github.com/drupal-composer/drupal-project), or some other way?  How did you install Drupal? How did you install Drush?

**Comment by Cswami1008** [comment]
Yes, I have a site-local Drush installation.  I installed Drupal using the recommended: "php composer.phar create-project drupal-composer/drupal-project:8.x-dev my_project --stability dev --no-interaction" boilerplate approach.  Before I was using the same except sans the php command and the .phar suffix because, as I understand, that only works when Composer is Global.

This boilerplate install automatically installs Drush, as far as I understand.  At least Drush appears in the vendor directory of the Drupal installation and in the C:\ProgramData\ directory as well.

I have completely ripped out my entire sites directory and all my Composer installs and re-installed just to see if any file was somehow damaged, but I always seem to get the same result.

I tried to use a different CLI tool, just to see if that was causing some type of alias of the path and that seemed to make no difference either.
